ExampleMod/Content/DamageClasses/ExampleDamageClass.cs

```diff
@@ -1,65 +1,68 @@
 			ClassName.SetDefault("example damage");
 		}
 
 		protected override float GetBenefitFrom(DamageClass damageClass) {
-			// Make this damage class not benefit from any otherclass stat bonuses by default, but still benefit from universal/all-class bonuses.
+			// This method lets you make your damage class benefit from other classes' stat bonuses by default, as well as universal stat bonuses.
+			// To briefly summarize the two nonstandard damage class names used by DamageClass:
+			// Default is, you guessed it, the default damage class. It doesn't scale off of any class-specific stat bonuses or universal stat bonuses.
+			// There are a number of items and projectiles that use this, such as thrown waters and the Bone Glove's bones.
+			// Generic, on the other hand, scales off of all universal stat bonuses and nothing else; it's the base damage class upon which all others that aren't Default are built.
 			if (damageClass == DamageClass.Generic)
 				return 1f;
 
-			// Now, I know you're just dyin' to get into the fun side of things, so let's have ourselves some demonstrations.
-			// Feel free to uncomment one of the below! Experiment, play with the variables a bit. See what works best for you!
+			return 0f;
+			// To explain how the return value works, it behaves like a percentage, with 0f being 0%, 1f being 100%, and so on.
+			// The return value indicates how much your class will scale off of any given damage class.
+			// For example, if I were to return 0.5f (50%) for DamageClass.Ranged, this custom class would receive all ranged stat bonuses at 50% effectiveness.
+			// There is no hardcap on what you can set this to.You can make it scale very heavily or barely at all...
+			// ...and you can even invert the scaling effect if you choose by returning a negative value.
 
-			// PROMPT: You want your damage class to benefit at a standard rate from a vanilla class' stat boosts.
-			// The below makes your class benefit at a standard rate (100%) from all melee stat bonuses.
-			/*
-			if (damageClass == DamageClass.Melee)
-				return 1f;
-			*/
+			// BONUS INFO:
+			// To refer to a non-vanilla damage class for these sorts of things, use "ModContent.GetInstance<YourDamageClassHere>()" instead of "DamageClass.XYZ".
+		}
 
-			// PROMPT: You want your damage class to benefit at a much higher rate from a vanilla class' stat boosts.
-			// The below makes your class benefit at 500% effectiveness from all magic stat bonuses.
-			/*
+		public override bool CountsAs(DamageClass damageClass) {
+			// This method allows you to make your damage class benefit from otherclass effects (e.g. Spectre bolts, Magma Stone) based on what returns true.
+			// Note that unlike GetBenefitFrom, you do not need to account for universal bonuses in this method.
+			// For this example, we'll make our class count as melee and magic for the purpose of class-specific effects.
+			if (damageClass == DamageClass.Melee)
```

Comment:
I get the feeling that someone may raise that this may be a place where being declarative might be better.

I might suggest considering if you can write it as ``Is 'damageClass' in {DamageClass.Melee, DamageClass.Magic}`` in all cases, or if this has a reason to be a method.

Reply:
personally, I can't see a significant reason as to why this would be better, as there's no way to modify existin' `DamageClass`es' return values on these methods (`GlobalDamageClass` doesn't --- and, if you ask me, probably doesn't need to --- exist) and, the way I see it, the way I have it set up already is both intuitive and versatile. if it ain't broke, don't fix it

Reply:
there is no conceivable way that this has enough of a performance impact to even remotely justify doin' it (even `GetBenefitFrom` isn't checked often enough per tick to present opportunity for noticeable improvement) and the amount of code necessary is already basically none. sorry, but I'm not convinced

Reply:
Alright, so, I guess to rephrase this a different way, I'd like to see the CountsAs vritual method replaced by a CountsAs public field consisting of List<damageClass>. 
The corresponding call to CountsAs can than be replaced with this.CountsAs.Contains(damageClass)

The advantages of this approach range from higher modifiability (and avoiding the need for a global), and performance gains in removing the performance overhead associated with virtual methods.

This is a fairly trivial change from a work involved perspective to push back on, is there a particular technical reason you're against changing it?